sdk/src/client/secret/types.rs

```diff
@@ -169,23 +169,24 @@ pub struct InputSigningDataDto {
     /// The output metadata
     pub output_metadata: OutputMetadataDto,
     /// The chain derived from seed, only for ed25519 addresses
-    pub chain: Option<Chain>,
+    pub chain: Option<Vec<u32>>,
 }
 
+#[allow(missing_docs)]
 impl InputSigningData {
-    pub(crate) fn try_from_dto(input: &InputSigningDataDto, token_supply: u64) -> Result<Self> {
+    pub fn try_from_dto(input: &InputSigningDataDto, token_supply: u64) -> Result<Self> {
         Ok(Self {
             output: Output::try_from_dto(&input.output, token_supply)?,
             output_metadata: OutputMetadata::try_from(&input.output_metadata)?,
-            chain: input.chain.clone(),
+            chain: input.chain.clone().map(Chain::from_u32_hardened),
```

Comment:
Is this correct? We consider everything to be hardened now? Which wasn't the case before?

Reply:
Before it was a Chain already
`from_u32_hardened` does the hardening